lnclient/second_ark/second_ark.go

```diff
@@ -0,0 +1,373 @@
+		return nil, err
+	}
+
+	var wallet *bindings.Wallet
+
+	dbFilePath := filepath.Join(workdir, barkDB)
+	if _, err := os.Stat(dbFilePath); err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			logger.Logger.WithError(err).Error("Failed to check Second Ark database file")
+			return nil, err
+		}
+
+		barkConfig := bindings.Config{
+			Network:              "signet",
+			Birthday:             nil,
+			AspAddress:           "https://ark.signet.2nd.dev",
+			EsploraAddress:       "https://esplora.signet.2nd.dev",
+			VtxoRefreshThreshold: 0,
+		}
+
+		wallet, err = checkBindingsErr(bindings.CreateWallet(dbFilePath, mnemonic, barkConfig))
+		if err != nil {
+			logger.Logger.WithError(err).Error("Failed to create Second Ark wallet")
+			return nil, err
+		}
+	} else {
+		wallet, err = checkBindingsErr(bindings.OpenWallet(dbFilePath, mnemonic))
+		if err != nil {
+			logger.Logger.WithError(err).Error("Failed to open Second Ark wallet")
+			return nil, err
+		}
+	}
+
+	logger.Logger.Info("Performing wallet maintenance")
+	if err := wallet.Maintenance().AsError(); err != nil {
+		logger.Logger.WithError(err).Error("Failed to perform wallet maintenance")
+		return nil, err
+	}
+
+	pk := wallet.OorPubkey()
+	logger.Logger.Info("Ark public key: ", pk)
+
+	cctx, cancel := context.WithCancel(ctx)
+	wg := &sync.WaitGroup{}
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		ticker := time.NewTicker(vtxoRefreshInterval)
+		defer ticker.Stop()
+
+		// Perform initial refresh on startup.
+		logger.Logger.Info("Refreshing vtxos")
+		if err := wallet.RefreshAll().AsError(); err != nil {
+			logger.Logger.WithError(err).Error("Failed to refresh vtxos")
+		}
+
+		for {
+			select {
+			case <-ticker.C:
+				logger.Logger.Info("Refreshing vtxos")
+				if err := wallet.RefreshAll().AsError(); err != nil {
+					logger.Logger.WithError(err).Error("Failed to refresh vtxos")
+				}
+			case <-cctx.Done():
+				return
+			}
+		}
+	}()
+
+	return &SecondArkService{
+		wallet: wallet,
+		cancel: cancel,
+	}, nil
+}
+
+func (s *SecondArkService) Shutdown() error {
+	logger.Logger.Info("Shutting down Ark client")
+
+	s.cancel()
+
+	logger.Logger.Info("Waiting for Ark procs to finish")
+	s.wg.Wait()
+
+	logger.Logger.Info("Ark shutdown complete")
+
+	return nil
+}
+
+func (s *SecondArkService) SendPaymentSync(ctx context.Context, invoice string, amount *uint64, timeoutSeconds *int64) (*lnclient.PayInvoiceResponse, error) {
+	paymentRequest, err := decodepay.Decodepay(invoice)
+	if err != nil {
+		logger.Logger.WithFields(logrus.Fields{
+			"bolt11": invoice,
+		}).WithError(err).Error("Failed to decode bolt11 invoice")
+
+		return nil, err
+	}
+
+	// Bark won't allow setting amount on invoices that already have an amount set.
+	// Bark also does not support msats, offering rounding up or down to the nearest sat.
+	// FIXME: Rounding up here to avoid underpayment; however, this may lead to unexpected spending.
```

Comment:
We could return an error if it's not a whole sat amount, I think that's sufficient for now

Reply:
If the amount is already in the invoice then the amount should not be provided separately right?